app/views/declaration/ducr_entry.scala.html

```diff
@@ -36,13 +36,19 @@
     insetText: exportsInsetText,
     inputWithTitleAsLabel: inputWithTitleAsLabel,
     formHelper: FormWithCSRF,
-    navigator: Navigator,
     appConfig: AppConfig
 )
 
 
 @(form: Form[Ducr])(implicit request: JourneyRequest[_], messages: Messages)
 
+@backLink = @{
+  request.cacheModel.ducr.fold(DucrChoiceController.displayPage) { ducr =>
```

Comment:
Can this logic not go in one of the navigator classes?

Reply:
For simple logic like this, and even more considering that is common to all journey types, honestly I do not see the point to involve the Navigator. IMO it would only make things more complicated.

Reply:
It seems like splitting responsibility for navigation by having some in the view and some in the navigator - I know we probably do this on other pages too - but it could be confusing for someone unfamiliar with the codebase.

Reply:
In any case you always go before to the page to figure out the key used for the Navigator's pattern matching.

Reply:
I agree, if we can put this logic in the Navigator class then we should to keep the logic in one place wherever possible :+1:

Reply:
And personally I would remove The "Common" pattern from the the Navigator. It seems to me we are over-engineering the Navigator just for the "responsibility"